src/common/CMakeLists.txt

```diff
@@ -152,50 +98,22 @@ if ("${CMAKE_SYSTEM_NAME}" STREQUAL "QNX")
 endif()
 target_compile_definitions(common_http PUBLIC MENDER_USE_BOOST_BEAST=${MENDER_USE_BOOST_BEAST})
 
-add_executable(http_test EXCLUDE_FROM_ALL http_test.cpp)
-target_link_libraries(http_test PUBLIC common_io common_http common_processes common_testing main_test gmock)
-gtest_discover_tests(http_test
-  ${MENDER_TEST_FLAGS}
-  WORKING_DIRECTORY ${MENDER_SRC_DIR}/common/http/testdata/
-  NO_PRETTY_VALUES
-)
-add_dependencies(tests http_test)
-
-add_executable(http_proxy_test EXCLUDE_FROM_ALL http_proxy_test.cpp)
-target_link_libraries(http_proxy_test PUBLIC common_http common_processes common_testing main_test gmock)
-gtest_discover_tests(http_proxy_test
-  ${MENDER_TEST_FLAGS}
-  WORKING_DIRECTORY ${MENDER_SRC_DIR}/common/http/testdata/
-  NO_PRETTY_VALUES
-)
-add_dependencies(tests http_proxy_test)
-
 add_library(common_log STATIC)
 # Accept the global compiler flags
 target_compile_options(common_log PRIVATE ${PLATFORM_SPECIFIC_COMPILE_OPTIONS})
 target_sources(common_log PRIVATE log/platform/boost/boost_log.cpp)
 target_link_libraries(common_log PUBLIC common_error Boost::log)
 
-# Test MenderLog
-add_executable(log_test EXCLUDE_FROM_ALL log_test.cpp)
-target_link_libraries(log_test PRIVATE common_log common_testing main_test gmock)
-gtest_discover_tests(log_test NO_PRETTY_VALUES)
-add_dependencies(tests log_test)
-
 add_library(common_config_parser STATIC config_parser/config_parser.cpp)
+if(BUILD_TESTS)
 target_link_libraries(common_config_parser PUBLIC common_json common_log gmock)
```

Comment:
Q: Why can we not add `gmock` to the `target_link_libraries` in the `tests` folder, and remove this if?